dace/sdfg.py

```diff
@@ -2317,6 +2317,51 @@ def traverse(node):
         # Return node that corresponds to current edge
         return traverse(tree_root)
 
+def trace_nested_access(node, state, sdfg):
+    """ Given an AccessNode in a nested SDFG, trace the accessed memory
+        back to the outermost scope in which it is defined.
+
+        :param node: An access node.
+        :param state: State in which the access node is located.
+        :param sdfg: SDFG in which the access node is located.
+        :return: A list of scopes (node, state, sdfg) in which the given
+                 data is accessed, from outermost scope to innermost scope.
+        """
+    if node.access == dace.dtypes.AccessType.ReadWrite:
+        raise NotImplementedError("Access node must be read or write only")
+    curr_node = node
+    curr_sdfg = sdfg
+    trace = [(node, state, sdfg)]
+    while curr_sdfg.parent is not None:
+        curr_state = curr_sdfg.parent
+        # Find the nested SDFG containing ourself in the parent state
+        for nested_sdfg in curr_state.nodes():
```

Comment:
missing an `else` clause (what if the node was not found?)

Reply:
Fixed with d082a61